### PR TITLE
upgrade amberdb.

### PR DESCRIPTION
**Pull request by shuangzhou**
This request upgrade amberdb from java 7 to java 8.  It also adds the loadWorks method to help with reporting of a list of diverse work records whether connected or not.

### CODE REVIEW

**Comment by shuangzhou** [comment]
@scoen @jrixon @ninhnguyen @m-r-c could you please review?  Thanks.

**Comment by nlabuilder** [comment]
SonarQube analysis reported 3 issues

1. MAJOR [AmberSession.java](https://github.com/nla/amberdb/blob/29e32f59b070116756fc04ea3fefd1354aed64ea/src/main/java/amberdb/AmberSession.java): This file has 1,113 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS00104)
1. MINOR [AmberSession.java#L325](https://github.com/nla/amberdb/blob/29e32f59b070116756fc04ea3fefd1354aed64ea/src/main/java/amberdb/AmberSession.java#L325): Define a constant instead of duplicating this literal "collection" 3 times. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS1192)
1. MINOR [AmberSession.java#L360](https://github.com/nla/amberdb/blob/29e32f59b070116756fc04ea3fefd1354aed64ea/src/main/java/amberdb/AmberSession.java#L360): Define a constant instead of duplicating this literal "bibId" 3 times. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS1192)

**Comment by nlabuilder** [comment]
Build finished.

File: src/main/java/amberdb/AmberSession.java [hunk 1]
```diff
@@ -1091,6 +1092,20 @@ private void increment(Map<String, Integer> countMap, String key) {
...
+     * be applied during reporting or mass work records export.  Please note this method does not
+     * traverse each work records from corresponding record ids, ie. not providing the children 
+     * records.
+     * @param ids are input record ids.
+     * @return stream of work records corresponding to the input record ids.
+     */
+    public Stream<Work> loadWorks(final List<Long> ids) {
+        AmberGraph g = getAmberGraph();
+        AmberQuery q = g.newQuery(ids);
+        return q.execute().stream().map(record -> graph.frame(record, Work.class));
```
Comment (m-r-c) [inline src/main/java/amberdb/AmberSession.java]: Nice

**Comment by m-r-c** [comment]
👍

**Pull request merged**